.github/workflows/flex_deploy_dev.yaml

```diff
@@ -59,22 +63,84 @@ jobs:
         run: |
           npm install
           npm run install-serverless-plugin
-          TWILIO_ACCOUNT_SID=${{ secrets.TWILIO_ACCOUNT_SID_DEV }} TWILIO_API_KEY=${{ secrets.TWILIO_API_KEY_DEV }} TWILIO_API_SECRET=${{ secrets.TWILIO_API_SECRET_DEV }} npm run deploy:$ENVIRONMENT
+          npm run deploy:$ENVIRONMENT
 
-  deploy-flex-config:
+  deploy-terraform:
     runs-on: ubuntu-latest
+    environment: dev
+    env:
+      ENVIRONMENT: dev
+      TWILIO_ACCOUNT_SID: ${{ secrets.TWILIO_ACCOUNT_SID }}
+      TWILIO_API_KEY: ${{ secrets.TWILIO_API_KEY }}
+      TWILIO_API_SECRET: ${{ secrets.TWILIO_API_SECRET }}
+      TF_VAR_TWILIO_ACCOUNT_SID: ${{ secrets.TWILIO_ACCOUNT_SID }}
+      TF_VAR_TWILIO_API_KEY: ${{ secrets.TWILIO_API_KEY }}
+      TF_VAR_TWILIO_API_SECRET: ${{ secrets.TWILIO_API_SECRET }}
     needs: [deploy-serverless, deploy-schedule-manager]
+    steps:
+      - uses: actions/checkout@v1 
+      - uses: ./.github/actions/terraform_state_artifact
+        with:
+          encryptionkey: ${{ secrets.TF_ENCRYPTION_KEY }}
+          path: ./infra-as-code/terraform/environments/default
+
+  ####################################################
+  ### NOTE: This job can be removed after the first 
+  ### successful deploy it is required to align
+  ### serverless hostname with terraform and then 
+  ### sids from terraform deploy back with serverless
+  #####################################################
```

Comment:
I think we can simplify this so that customers don't need to remove it to save deploy time. Either the terraform job could output some piece of state or we could have the CLI pull the env vars for the service. Then, this job could use an `if` statement to guard its execution.